tools/src/main/scala/scala/scalanative/build/Build.scala

```diff
@@ -54,32 +54,32 @@ object Build {
     config.logger.time("Total") {
       val fclasspath = NativeLib.filterClasspath(config.classPath)
       val fconfig    = config.withClassPath(fclasspath)
+      val workdir    = fconfig.workdir
 
-      val workdir = fconfig.workdir
+      // create optimized code and generate ll
       val entries = ScalaNative.entries(fconfig)
       val linked  = ScalaNative.link(fconfig, entries)
       ScalaNative.logLinked(fconfig, linked)
       val optimized = ScalaNative.optimize(fconfig, linked)
-
-      // clean ll files
-      IO.getAll(workdir, "glob:**.ll").foreach(Files.delete)
-
       val generated = ScalaNative.codegen(fconfig, optimized)
 
-      val nativelibs   = NativeLib.findNativeLibs(fconfig.classPath, workdir)
-      val nativelib    = NativeLib.findNativeLib(nativelibs)
-      val unpackedLibs = nativelibs.map(LLVM.unpackNativeCode(_))
-
       val objectPaths = config.logger.time("Compiling to native code") {
-        val nativelibConfig =
-          fconfig.withCompilerConfig(
-            _.withCompileOptions("-O2" +: fconfig.compileOptions))
-        val libObjectPaths =
-          LLVM.compileNativelibs(nativelibConfig,
-                                 linked,
-                                 unpackedLibs,
-                                 nativelib)
+        // find native libs
+        val nativelibs = NativeLib.findNativeLibs(fconfig.classPath, workdir)
+
+        // compile all libs
+        val libObjectPaths = nativelibs
+          .map(nl => NativeLib.unpackNativeCode(nl))
+          .map(destPath => {
+            val paths = NativeLib.findNativePaths(workdir, destPath)
+            Filter.filterNativelib(fconfig, linked, destPath, paths)
+          })
+          .map(filteredPaths => LLVM.compile(fconfig, filteredPaths))
```

Comment:
Yes, that was my idea exactly. I think 

I think making the solution very general may be hard but for now the solution can be private to Scala Native libraries. I was unaware that Windows and BSD socket were partly compatible. Since the projects are all handled independently then we can have the properties file named the same.

```scala
 /** To find filter file */
  private val filterProperties = s"${codeDir}-filter.properties"
```
If we need some logic based on the project just to make things easier we could put the project name in the properties file as well - we could use it to report on filtering or maybe compiling errors as well?

```
project=org.scala-native.posixlib
```

Reply:
Maybe it might be a good idea to rename it from `-filter.properties` to `-scala-native.properties` or `-sn.properties` for short. This way we can make it more generic and add additional stuff to that later if needed. 

One additional use I can think of is a combination of project name and version info, that might be used to resolve transitive dependencies in favour of the latest one, eg., if for some reason we would have two versions of nativelib. In such case, it might fail at linker, due to duplicated definitions. I actually had such a problem with locally build sources which lead me to create PR #2207

Reply:
The directory name is `scala-native` so I think that `scala-native.properties` is a great idea and lines up nice. Great suggestion.